sheepdog/blueprint/routes/views/__init__.py

```diff
@@ -112,13 +112,22 @@ def root_create():
             "dbgap_accession_number": "phs000178"
         }
     """
-    message = None
-    node_id = None
-    doc = parse.parse_request_json()
+    input_doc = flask.request.get_data().decode("utf-8")
+    content_type = flask.request.headers.get("Content-Type", "").lower()
+    if content_type == "text/csv":
+        doc, _ = utils.transforms.CSVToJSONConverter().convert(input_doc)
+    elif content_type in ["text/tab-separated-values", "text/tsv"]:
+        doc, _ = utils.transforms.TSVToJSONConverter().convert(input_doc)
+    else:
+        doc = utils.parse.parse_request_json()
```

Comment:
If possible, a minor suggestion would be to check if any errors are returned from the `convert` calls.